pagefaultgames/pokerogue
# [Bug] Fixes bug with frenzy moves keeping the confusion counter despite disruption

## What are the changes?

Frenzy moves (Outrage, Thrash, Petal Dance, Raging Fury) will maintain the confusion counter when disrupted, causing confusion immediately when calling the move again.
E.g. Use outrage on opponent and 1HKO that opponent, enemy switches in fairy type (outrage disrupted). Use outrage again later on and you will be immediately confused after one use.

## Why am I doing these changes?



This is a bug with all frenzy moves

## What did change?


Added the NoEffectAttr with frenzyMissFunc to each of the frenzy moves to lapse the Frenzy tag when a move has no effect. 
Refactored the FrenzyAttr apply function to actually lapse the tag each turn. 
Account for disruption by removing the tag in frenzyMissFunc
Removes ```tag.turnCount = 0``` in removeTag, since other functions do not rely on turnCount when being removed, like Frenzy does.

### Screenshots/Videos


Before (see [conversation](https://github.com/pagefaultgames/pokerogue/pull/3041#issuecomment-2231768795))
After (3 turn Outrage - disrupted with no confusion, followed by 2 turn Outrage - disrupted with confusion)

https://github.com/user-attachments/assets/80a45caf-f2fa-43ed-a77c-a198d505b576

## How to test the changes?




Give opponent Outrage, give yourself a non-Fairy mon and a Fairy mon. Let the opponent use outrage on you once, switch into Fairy type, switch back to original mon and let opponent use Outrage again. See change.

## Checklist
